packages/mattermost-redux/src/reducers/entities/files.ts

```diff
@@ -131,6 +150,25 @@ export function fileIdsByPostId(state: Record<string, string[]> = {}, action: Ge
     }
 }
 
+function storeAllFilesIdsForPost(state: Record<string, string[]>, post: Post) {
```

Comment:
This is very similar to `storeAllFilesForPost`, I wonder if we can just use one function?

Reply:
Yeah both these functions are the same, only difference is that `state` is of a different type in each. I looked at doing a type check and overloading `storeAllFilesForPost` to handle both but I couldn't find an efficient or readable solution that would work in this case. Main issue stems from `state` being of type `Record` which makes it difficult to check the type of its objects at run time.